x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/hooks/use_fetch_data_views.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useState, useEffect } from 'react';
+import { DataViewBase } from '@kbn/es-query';
+import { BASE_RAC_ALERTS_API_PATH } from '../../../../../../rule_registry/common';
+import { DataPublicPluginStart } from '../../../../../../../../src/plugins/data/public';
+import { HttpSetup } from '../../../../../../../../src/core/public';
+import { Consumer } from '../types';
+
+export function useFetchDataViews(
+  consumers: Consumer[],
+  http: HttpSetup,
+  data: DataPublicPluginStart
+) {
+  const [dataViews, setDataViews] = useState<DataViewBase[]>([]);
+  useEffect(() => {
```

Comment:
This fetch is missing a cleanup function and abort controller.

Reply:
nice catch, will add